docs/proposals/20210705-bastion-vms.md

```diff
@@ -0,0 +1,209 @@
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Use Machine Deployments to bootstrap bastion VMs for SSH/RDP access to k8s nodes.
+
+## Table of Contents
+
+* [Use Machine Deployments to bootstrap bastion VMs for SSH/RDP access to k8s nodes.](#use-machine-deployments-to-bootstrap-bastion-vms-for-sshrdp-access-to-k8s-nodes)
+  * [Table of Contents](#table-of-contents)
+  * [Glossary](#glossary)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals/Future Work](#non-goalsfuture-work)
+  * [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Story 1](#story-1)
+      * [Functional Requirements](#functional-requirements)
+      * [Non-Functional Requirements](#non-functional-requirements)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Existing APIs for Clarity](#existing-apis-for-clarity)
+      * [Proposed API Changes](#proposed-api-changes)
+  * [Conclusions](#conclusions)
+  * [Additional Details](#additional-details)
+    * [Test Plan](#test-plan)
+  * [Implementation History](#implementation-history)
+
+
+## Glossary
+
+* CAPZ - An abbreviation of Cluster API Provider Azure.
+* KISS - Keep It Simple, Stupid (see [Wikipedia](https://en.wikipedia.org/wiki/KISS_principle))
+
+## Summary
+
+Accessing virtual machines forming a Kubernetes cluster for administrative purposes is a potential security threat.
+A common pattern to provide SSH/RDP access to virtual machines without exposing them to the public internet is to use 
+a `Bastion host` or `Jump host`, a dedicated machine that is exposed to the internet acting as a bridge between the administrators'
+computers and the Kubernetes nodes (that can therefore stay behind a firewall).
+This CAEP aims at providing a way to bootstrap `Bastion Hosts` in a CAPZ cluster levereging existing Controllers as much as possible
+by reusing CRDs like `MachineDeployment` and `AzureMachineTemplate`.
+
+## Motivation
+
+By default CAPZ clusters expose one of the Master nodes to the internet making it possible to access all the other nodes in the cluster via SSH or RDP.
+
+As an alternative, it is possible to use the [`AzureBastion`](https://azure.microsoft.com/en-us/services/azure-bastion/) feature to SSH/RDP to nodes using a web interface through 
+the `Azure Portal` (see [documentation](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/docs/book/src/topics/ssh-access.md)).
+
+Those solutions might not cover all use cases. As a matter of example:
+
+- Using one master node as a jump host might be not compliant to the security policies for some companies.
+- Using `AzureBastion` might be inpractical or unusable in some use cases (machine to machine access would be nearly impossible).
+
+### Goals
+
+- Allow users to deploy virtual machines as bastion hosts in multiple availability zones using a custom cloud init file.
+
+### Non-Goals/Future Work
```

Comment:
should we also add removing the existing AzureBastion service support as a non-goal?

Reply:
I'd say no. In my vision `AzureBastion` has a different use case and it can be an alternative solution to whom that don't need/want a dedicated VM as bastion.